content/blog/2023/07/16/2023-07-16-unreleased-production-changes-probe.adoc

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: Unreleased Production Changes Probe
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/blog_20230702.png
+sig:
+- gsoc
+---
+
+== Introduction
+
+A bug fix, security fix, or new features are of no use unless they are released.
+
+A production release not only provides users with an updated plugin, but also gives the developers a chance to get feedback on their work.
+
+The "unreleased production changes" probe looks for unreleased commits in the production files.
+
+If an unreleased file is found in the production path, the probe returns a failure message.
+If no unreleased file is found, it returns a success message.
+
+NOTE: For now, we are focusing on Maven production files (`pom.xml` and `src/main` folder). 
+
+This is my first probe that went live as a GSoC contributor.
+
+== Importance of the probe
+The probe aims to identify unreleased production commits in  Maven production files.
+
+Finding an unreleased production file can mean one or more of the following:
+
+* There might be unreleased security fixes that should be released as soon as possible.
+* There might be unreleased features requested by active users to improve the usefulness of the plugin.
+* The plugin maintainer may be inactive or too busy to further develop the plugin.
+This can indicate the plugin needs a new maintainer, and it should be up for adoption.
```

Comment:
```suggestion
* There are unreleased security fixes that should be released as soon as possible.
* There are unreleased features requested that would improve the usefulness of the plugin.
* The plugin maintainer may be inactive or too busy to further develop the plugin.
This could indicate the plugin needs a new maintainer and should be up for adoption.
```
suggestion for sentence wording